azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java

```diff
@@ -1605,6 +1617,20 @@ private String getPodName(final int executionId) {
     return String.join("-", this.podPrefix, this.clusterName, String.valueOf(executionId));
   }
 
+  /**
+   * A helper method to generate a k8s service name of the given execution ID.
+   * @param serviceNamePrefix prefix of the k8s service
+   * @param clusterName current cluster's name
+   * @param executionId ID of the execution on the pod
+   * @return the k8s service name of the given execution ID
+   */
+  public static String getServiceNameHelper(
```

Comment:
If you want to generialize the name generation helper, why not include getPodName too, just serviceNameHelper for getServiceName isn't necessary to add additional layer on unless it is more generic.

Reply:
that's a good point - did not include getPodName since it is used by others, but let me add it for consistency.

Reply:
I actually think it can be simpfied into one method, as they both are just String.join using "-", just a NamePrefix change.
the helper function would be just String.join("-", prefix, clusterName, execId). 
So I feel it would make the helper function into a real reusable helper. : )
Sorry I didn't make it super clear at the beginning.
You can disagree it is just a matter of format, people have different opinion of what a clean code looks like.

Reply:
yup that makes sense. Updated the PR and now we only have a static helper method `getK8sEntityName` to return the pod, service and VPA name.